### ISSUE TITLE
Using `ignore` error strategy doesn't respect `debug = true` for failed processes

### ISSUE THREAD

**Issue by ramdayan** [nextflow-io/nextflow#4672]
### Expected behavior and actual behavior

When using the `ignore` error strategy in combination of setting `debug` to `true`, I expect to see the output of both succeeding and failing processes. However, the output of failing processes is omitted.

### Steps to reproduce the problem

```
#!/usr/bin/env nextflow

nextflow.enable.dsl = 1

process p1 {
    debug true
    errorStrategy 'ignore'

    script:
    """
    #!/usr/bin/env bash
    set -e

    sleep 1
    echo "Success!"
    """
}

process p2 {
    debug true
    errorStrategy 'ignore'

    script:
    """
    #!/usr/bin/env bash
    set -e

    echo "Failing..."
    exit 1
    """
}
```

### Program output 

```
N E X T F L O W  ~  version 22.10.8
Launching `main.nf` [hopeful_marconi] DSL1 - revision: 9e64aa2e64
executor >  local (2)
[73/c2f38f] process > p1 [100%] 1 of 1 ✔
[0e/79f75d] process > p2 [100%] 1 of 1, failed: 1 ✔
Success!

[0e/79f75d] NOTE: Process `p2` terminated with an error exit status (1) -- Error is ignored
```

### Environment 

* Nextflow version: 22.10.8 
* Java version: 17.0.5
* Operating system: macOS
* Bash version: 3.2.57

### Additional context

Although this example is in DSL 1, it probably affects DSL 2 scripts as well.

**Comment by bentsherman** [comment]
This is the expected behavior, task stdout is printed only if the task succeeded, or as part of the standard task error message when the pipeline is terminated

**Comment by ramdayan** [comment]
I must say this is very annoying when trying to understand what went wrong in a pipeline.
What is the reasoning behind omitting the output of failed tasks?
And why is it justified then to print out their output if the pipeline was terminated?

**Comment by bentsherman** [comment]
The intention of the ignore strategy is that you are expecting a few tasks to fail, but you want the pipeline to process as much as it can and then review any failures afterwards. If you're wanting to immediately see why the task is failing, that suggests that you are still debugging issues in your pipeline code, in which case you should use "terminate" or "finish" to also end the pipeline immediately (and print the task error).

You can also get the task stdout/stderr from the work directory while the workflow is running, if you want to keep using "ignore" but also see the error immediately

**Comment by ramdayan** [comment]
I don't need to see why the task is failing immediately, but even after the pipeline has finished, it would be much more convenient to see the failing tasks output in the pipeline's output rather than to dig into the work dir to find the output of the failing tasks.

**Comment by bentsherman** [comment]
Fair enough. You can also use the [nextflow log](https://nextflow.io/docs/latest/tracing.html#execution-log) command to query information about the tasks. Here's an example that will show the stderr of every failed task in the previous run:
```bash
nextflow log last -f hash,workdir,stderr -F 'exit != 0'
```

**Comment by bentsherman** [comment]
Closing in favor of #903 and #7233
